lib/recurly/pricing/checkout/index.js

```diff
@@ -210,30 +246,69 @@ export default class CheckoutPricing extends Pricing {
         }
       }
 
-      let adjustment = {};
-
-      if (existingAdjustment) {
-        // Only pick present attributes
-        if (typeof amount !== 'undefined') adjustment.amount = amount;
-        if (typeof currency !== 'undefined') adjustment.currency = currency;
-        if (typeof quantity !== 'undefined') adjustment.quantity = parseInt(quantity, 10);
-        if (typeof taxExempt !== 'undefined') adjustment.taxExempt = taxExempt;
-        if (typeof taxCode !== 'undefined') adjustment.taxCode = taxCode;
-
-        Object.assign(existingAdjustment, adjustment);
-      } else {
-        // Adjustment defaults
-        quantity = parseInt(quantity, 10);
-        if (isNaN(quantity)) quantity = 1;
-        currency = currency || this.items.currency;
-        taxExempt = !!taxExempt;
+      if (existingAdjustment && existingAdjustment.itemCode && !itemCode) {
+        itemCode = existingAdjustment.itemCode;
+      }
```

Comment:
To demonstrate what this is meant to protect:
```js
const checkoutPricing = recurly.Pricing.Checkout();
checkoutPricing.adjustment({ itemCode: 'my-item-code', id: 'adjustment-0' });
// ...
// I'd like to override my item price
checkoutPricing.adjustment({ id: 'adjustment-0', amount: 20 });
```

Subsequent routines in this method will use `itemCode` as a sentinel, so we must promote the value here.